OpenRA.Mods.Common/UpdateRules/Rules/AddResourceRenderer.cs

```diff
@@ -0,0 +1,65 @@
+﻿#region Copyright & License Information
+/*
+ * Copyright 2007-2019 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Linq;
+
+namespace OpenRA.Mods.Common.UpdateRules.Rules
+{
+	public class AddResourceRenderer : UpdateRule
+	{
+		public override string Name
+		{
+			get { return "Add ResourceRenderer trait"; }
+		}
+
+		public override string Description
+		{
+			get { return "The rendering parts of ResourceLayer have been moved to a new trait"; }
+		}
+
+		readonly List<string> locations = new List<string>();
+
+		public override IEnumerable<string> AfterUpdate(ModData modData)
+		{
+			if (locations.Any())
+				yield return "[D2k]ResourceRenderer has been added.\n" +
+					"You need to adjust the the field RenderTypes on trait [D2k]ResourceRenderer\n" +
+					"on the following actors:\n" +
+					UpdateUtils.FormatMessageList(locations);
+
+			locations.Clear();
+		}
+
+		public override IEnumerable<string> UpdateActorNode(ModData modData, MiniYamlNode actorNode)
+		{
+			if (!actorNode.ChildrenMatching("ResourceLayer").Any() && !actorNode.ChildrenMatching("ResourceRenderer").Any())
```

Comment:
Doesn't this have to be `actorNode.ChildrenMatching("ResourceLayer").Any() && !actorNode.ChildrenMatching("ResourceRenderer").Any()`?